### PR TITLE
fix: Swap-Send flaky test fix

### LINKED ISSUE #25026: flaky test: "Swap-Send ERC20 to non-contract address with data that matches swap data signature submits a transaction successfully"
### What is this about?

Failure:
https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/85136/workflows/e28f8cf1-47cd-47bc-8874-f729489b18ac/jobs/3080487/tests

Error message

Image

Screenshot

Image

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by vthomas13**
## **Description**

This PR is a fix for the flakiness in the e2e test case: "Swap-Send ERC20 to non-contract address with data that matches swap data signature submits a transaction successfully"
image
Additionally there are some minor readability improvements.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25041?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-extension/issues/25026

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/webdriver/driver.js [hunk 1]
```diff
@@ -63,12 +63,6 @@ function wrapElementWithAPI(element, driver) {
     }
   };
 
-  element.nestedFindElement = async (rawLocator) => {
```
Comment (vthomas13) [inline test/e2e/webdriver/driver.js:66]: I think this is somewhat inconsistent with the other webdriver methods for finding elements [findElement(s), findClickableElement(s), findVisibleElement, etc.]

File: test/e2e/webdriver/driver.js [hunk 2]
```diff
@@ -465,6 +459,20 @@ class Driver {
...
 
+  /**
+   * Finds a nested element within a parent element using the given locator.
+   * This is useful when the parent element is already known and you want to find an element within it.
+   *
+   * @param {WebElement} element - Parent element
+   * @param {string | object} nestedLocator - Nested element locator
+   * @returns {Promise<WebElement>} A promise that resolves to the found nested element.
+   */
+  async findNestedElement(element, nestedLocator) {
```
Comment (vthomas13) [inline test/e2e/webdriver/driver.js:470]: Same file, slightly different name, added docs

**Review by DDDDDanica — APPROVED** [review]

**Comment by DDDDDanica** [comment]
LGTM !

**Review by chloeYue — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**